### PR TITLE
IDEX-2983: fixed NativeGitConnection class to throw correct exception when there is no any remote repository

### PR DESCRIPTION
**Pull request by dmytro-ndp**
Please, review fix and merge changes into the master.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@sleshchenko  @akorneta  wdyt?

File: platform-api/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LsRemoteTest.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void testShouldBeAbleToGetResultFromPublicRepo(GitConnectionFactory conne
 
     @Test(dataProvider = "GitConnectionFactory", dataProviderClass = org.eclipse.che.git.impl.GitConnectionFactoryProvider.class,
           expectedExceptions = UnauthorizedException.class,
-          expectedExceptionsMessageRegExp = "Not authorized")
+          expectedExceptionsMessageRegExp = "fatal: Authentication failed for 'https://bitbucket.org/exoinvitemain/privater.git/'\n")
```
Comment (garagatyi) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LsRemoteTest.java]: What if locale is not english? Will this message be the same?

Comment (dmytro-ndp) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LsRemoteTest.java]: I don't know.

File: che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/NativeGitConnection.java [hunk 2]
```diff
@@ -432,20 +396,29 @@ public PullResponse pull(PullRequest request) throws GitException, UnauthorizedE
         return pullCommand.getPullResponse();
     }
 
-    @Override
-    public PushResponse push(PushRequest request) throws GitException, UnauthorizedException {
-        ensureExistenceRepoRootInWorkingDirectory();
-        String remoteUri;
+    private String getRemoteUri(String remoteName) throws GitException {
```
Comment (sleshchenko) [inline che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/NativeGitConnection.java]: I think it better to move private method to below of all public methods

File: platform-api/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LsRemoteTest.java (hunk 1 above)
Comment (sleshchenko) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/git/impl/LsRemoteTest.java]: For executing of command is always used english locale https://github.com/codenvy/che-core/blob/master/che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/CommandProcess.java#L54

File: che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/NativeGitConnection.java (hunk 2 above)
Comment (dmytro-ndp) [inline che-core-git-impl-native/src/main/java/org/eclipse/che/git/impl/nativegit/NativeGitConnection.java]: You are right.

**Comment by sleshchenko** [comment]
Looks good to me :+1:

**Comment by garagatyi** [comment]
LGTM

**Comment by akorneta** [comment]
+1

**Comment by skabashnyuk** [comment]
Go

**Comment by dmytro-ndp** [comment]
Thank you for review, guys!

**Comment by dmytro-ndp** [comment]
Changes have been merged into master with commit ec0d202441a8bda5ab900401e39e845f99ae8573

**Comment by skabashnyuk** [comment]
Don't forget to merge in 4.0

**Comment by dmytro-ndp** [comment]
OK, I remember.

**Pull request closed without merging**